hbase-replication/src/main/java/org/apache/hadoop/hbase/replication/TableReplicationQueueStorage.java

```diff
@@ -532,4 +532,13 @@ public Set<String> getAllHFileRefs() throws ReplicationException {
       throw new ReplicationException("failed to getAllHFileRefs", e);
     }
   }
+
+  @Override
+  public boolean hasData() throws ReplicationException {
+    try {
+      return conn.getAdmin().getDescriptor(tableName) != null;
```

Comment:
Oh, here we can only use Admin...

Anyone, can be a follow on issue to see if we can use master services directly.

Reply:
@Apache9  sir
Well, we can only use admin here, unless the masterServices  is passed in the constructor of TableReplicationQueueStorage,so do we need to optimize further?